src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -184,13 +223,49 @@ private List<ProgrammingExerciseTask> extractTasks(ProgrammingExercise exercise)
             var testCaseNames = extractTestCaseNames(capturedTestCaseNames);
 
             for (String testName : testCaseNames) {
```

Comment:
To solve the Teamscale warning which correctly points out the many indentations, how about extracting the for loop in a helper method that returns a `List<ProgrammingExerciseTestCase>` which you then can add with `tasks.getTestCases().addAll(...)`